crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -352,37 +370,57 @@ impl RoomEventCacheInner {
     }
 
     #[instrument(skip_all, fields(room_id = %self.room_id))]
-    pub(super) async fn handle_joined_room_update(&self, updates: JoinedRoomUpdate) -> Result<()> {
-        self.handle_timeline(
-            updates.timeline,
-            updates.ephemeral.clone(),
-            updates.ambiguity_changes,
-        )
-        .await?;
+    pub(super) async fn handle_joined_room_update(
+        &self,
+        updates: JoinedRoomUpdate,
+        generic_room_event_cache_update_sender: &Sender<GenericRoomEventCacheUpdate>,
+    ) -> Result<()> {
+        if self
+            .handle_timeline(updates.timeline, updates.ephemeral.clone(), updates.ambiguity_changes)
+            .await?
+        {
+            let _ = generic_room_event_cache_update_sender.send(
+                GenericRoomEventCacheUpdate::TimelineUpdated { room_id: self.room_id.clone() },
+            );
+        }
 
         self.handle_account_data(updates.account_data);
 
         Ok(())
     }
 
     #[instrument(skip_all, fields(room_id = %self.room_id))]
-    pub(super) async fn handle_left_room_update(&self, updates: LeftRoomUpdate) -> Result<()> {
-        self.handle_timeline(updates.timeline, Vec::new(), updates.ambiguity_changes).await?;
+    pub(super) async fn handle_left_room_update(
+        &self,
+        updates: LeftRoomUpdate,
+        generic_room_event_cache_update_sender: &Sender<GenericRoomEventCacheUpdate>,
+    ) -> Result<()> {
+        if self.handle_timeline(updates.timeline, Vec::new(), updates.ambiguity_changes).await? {
+            let _ = generic_room_event_cache_update_sender.send(
+                GenericRoomEventCacheUpdate::TimelineUpdated { room_id: self.room_id.clone() },
+            );
+        }
+
         Ok(())
     }
 
+    /// Handle a [`Timeline`], i.e. new events receivedy by a sync for this
+    /// room.
+    ///
+    /// It returns `Ok(true)` if events have been added, redacted or removed,
+    /// `Ok(false)` otherwise.
     async fn handle_timeline(
```

Comment:
Is there a test for this function? If it can be added easily it seems worth it now we have the boolean return value that we need to get right.

Reply:
Resolved since this method no longer returns a boolean 😛.